runtime/j9vm/java11vmi.c

```diff
@@ -457,16 +458,27 @@ static BOOLEAN
 areNoPackagesDefined(J9VMThread * currentThread, J9ClassLoader * classLoader, const char* const* packages, U_32 numPackages)
 {
 	BOOLEAN success = TRUE;
+	J9JavaVM * vm = currentThread->javaVM;
+	J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+
+	 /*
+	 * This check will be ignored for calls to this method that occur before java.base is defined. 
+	 * Classes are loaded before java.base is created that are added to the classHashTable. 
+	 * These classes will eventually be fixed up to be part of java.base, but should not be considered duplicate packages
+	 * before that happens.
+	 */
+	BOOLEAN checkDefinedPackages = J9_ARE_ALL_BITS_SET(vm->runtimeFlags, J9_RUNTIME_JAVA_BASE_MODULE_CREATED);
 
 	if (NULL != packages) {
 		U_32 const arrayLength = numPackages;
 		if (0 != arrayLength) {
 			U_32 i = 0;
-			for (i = 0; i < arrayLength; i++) {
+			for (i = 0; success && (i < arrayLength); i++) {
 				const char *packageName = packages[i];
-				if (isPackageDefined(currentThread, classLoader, packageName)) {
+				if (checkDefinedPackages
+						&& vmFuncs->isAnyClassLoadedFromPackage(classLoader, (U_8*) packageName, strlen(packageName))
```

Comment:
Changing from `isPackageDefined` -> `isAnyClassLoadedFromPackage` - have you confirmed this gives the same results?  My read of the two functions indicates they'd give different results if the module hasn't loaded a class in the package yet.  Basically, two modules could "claim" `package.A` with this code provided they hadn't loaded a class in it yet, splitting the package across the modules.

Do you have a test case for this case?

Reply:
@theresa-m I was circling back to this as I thought I saw similar code in a JDK 15 PR.  Can you confirm the test case for this?

Reply:
I tried your suggested scenario. OpenJ9 does throw the correct error, but as you predicted it's not in this line of code.

The case is covered later when class loaders are initialized:
```
java.lang.LayerInstantiationException: Package com.greetings in both module org.astro and module com.greetings
	at java.base/jdk.internal.module.ModuleBootstrap.checkSplitPackages(ModuleBootstrap.java:499)
	at java.base/jdk.internal.module.ModuleBootstrap.boot(ModuleBootstrap.java:406)
	at java.base/java.lang.ClassLoader.initializeClassLoaders(ClassLoader.java:211)
	at java.base/java.lang.Thread.initialize(Thread.java:430)
	at java.base/java.lang.Thread.<init>(Thread.java:155)
```

Reply:
Hey @DanHeidinga do you think the above error is sufficient to resolve this?

Reply:
Does it match the RI's behaviour?

Reply:
Yes it matches.

Reply:
@DanHeidinga I think this is ready to go unless there are other concerns.